src/main/java/org/antlr/intellij/plugin/psi/LexerRuleRefNode.java

```diff
@@ -1,15 +1,16 @@
 package org.antlr.intellij.plugin.psi;
 
-import com.intellij.psi.PsiReference;
 import com.intellij.psi.tree.IElementType;
+import com.intellij.psi.util.PsiTreeUtil;
 
 public class LexerRuleRefNode extends GrammarElementRefNode {
 	public LexerRuleRefNode(IElementType type, CharSequence text) {
 		super(type, text);
 	}
 
 	@Override
-	public PsiReference getReference() {
-		return new GrammarElementRef(this, getText());
+	protected boolean isDeclaration() {
```

Comment:
I would have kept `getReference()` here and simply add this snippet:
```
		if (getParent() instanceof LexerRuleSpecNode || getParent() instanceof TokenSpecNode) {
			// A declaration doesn't need to reference itself
			return null;
		}
```

(same in `ParserRuleRefNode`)